### PR TITLE
wgengine/magicsock: prefer IPv6 transport if roughly equivalent latency

### LINKED ISSUE #1566: wgengine/magicsock: prefer IPv6 as transport if roughly equivalent latency as IPv4
Sometimes IPv4 behind NATs doesn't have a stable port (e.g. #1559). If two peers both have working IPv6, they should prefer that over IPv4, even if the IPv4 disco probe might've come in 1ms earlier. If IPv6 is within the same ballpark latency wise, we should use IPv6.

/cc @tailscale/packets

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes #1566

### CODE REVIEW

File: wgengine/magicsock/magicsock.go [hunk 1]
```diff
@@ -3670,6 +3670,17 @@ func betterAddr(a, b addrLatency) bool {
 	if a.IsZero() {
 		return false
 	}
+	if a.IP.Is6() && b.IP.Is4() {
+		// Prefer IPv6 for being a bit more robust, as long as
+		// the latencies are roughly equivalent.
+		if a.latency/10*9 < b.latency {
```
Comment (danderson) [inline wgengine/magicsock/magicsock.go:3676]: I don't understand this comparison. Why are you dividing one side and not the other? How does that translate to "roughly equivalent" ?

Comment (bradfitz) [reply wgengine/magicsock/magicsock.go:3676]: They're in the same ballpark latency-wise from our single data point. It's not 10% worse.

Comment (bradfitz) [reply wgengine/magicsock/magicsock.go:3676]: There are test cases with comments too.

Comment (danderson) [reply wgengine/magicsock/magicsock.go:3676]: Oh, I misread that as /(10^9), and parsed it as "nanoseconds to seconds conversion", at which point the compare makes no sense.

Okay, yeah, now I'm reading it right, LGTM.

**Review by danderson — APPROVED** [review]

**Pull request merged**